### DISCUSSION TITLE
Support for templating in parcel 2

### DISCUSSION THREAD

**Discussion by alexsoin** [parcel-bundler/parcel#7277] [Q&A]
Do you plan to add template engine support to the parcel? For example: [nunjucks](https://mozilla.github.io/nunjucks/), [twig](https://twig.symfony.com/), [smarty](https://www.smarty.net/). 

Found a plugin https://www.npmjs.com/package/parcel-plugin-nunjucks but as I understand it is not supported in parcel 2.

Or maybe it is possible to send a file for rendering to a js plugin?

**Comment by mischnic** [comment]
This is a Parcel 2 plugin for EJS: https://github.com/sarscode/parcel-transformer-ejs

**Reply by christian-beckmann** [reply]
@alexsoin maybe this helps https://github.com/christian-beckmann/parcel-transformer-twig (wip)

**Reply by KaKi87** [reply]
> This is a Parcel 2 plugin for EJS

@mischnic
How can I use it with with the programmatic API ?
Here's my current code :
```js
new Parcel({
    entries: 'src/index.html',
    defaultConfig: '@parcel/config-default',
    env: { NODE_ENV: 'development' },
    serveOptions: { port: config.devServerPort },
    hmrOptions: { port: config.devServerPort }
})
```
> maybe this helps [christian-beckmann/parcel-transformer-twig](https://github.com/christian-beckmann/parcel-transformer-twig) (wip)

@christian-beckmann
Unfortunately I'm running Node 14.x, not 16.x

Thanks

**Comment by christian-beckmann** [comment]
I did some improvements, but it should run (with a little workaround with .twigrc watchfolder property). The problem with the v1 twigjs plugin was, that it cached all included/embeded https://github.com/twigjs/twig.js/issues/794, so the twig.js cache is disabled by default now, to prevent these problems. The watchfolder property is a workaround to inform parcel about all twig files in a folder, to keep track of changes - because I still found no way to get information about included/embeded files by twig.js itself.

**Comment by KaKi87** [comment]
> For example: [nunjucks](https://mozilla.github.io/nunjucks/), [twig](https://twig.symfony.com/), [smarty](https://www.smarty.net/)

There's also [Eta](https://eta.js.org/) and [EJS](https://ejs.co/).

**Reply by christian-beckmann** [reply]
Sry, change it to 12, like in parcel 2, package.json defined.
